aws/table_aws_cloudwatch_metric_data_point.go

```diff
@@ -0,0 +1,280 @@
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "timestamp",
+				Description: "The timestamp for the data points, formatted in Unix timestamp format.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "value",
+				Description: "The data point for the metric corresponding to Timestamp.",
+				Type:        proto.ColumnType_DOUBLE,
+			},
+			{
+				Name:        "expression",
+				Description: "This field can contain either a Metrics Insights query, or a metric math expression to be performed on the returned data.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromQual("expression"),
+			},
+			{
+				Name:        "scan_by",
+				Description: "The order in which data points should be returned. TimestampDescending returns the newest data first and paginates when the MaxDatapoints limit is reached. TimestampAscending returns the oldest data first and paginates when the MaxDatapoints limit is reached.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromQual("scan_by"),
+			},
+			{
+				Name:        "timezone",
+				Description: "You can use timezone to specify your time zone so that the labels of returned data display the correct time for your time zone.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromQual("timezone"),
+			},
+			{
+				Name:        "metric_stat",
+				Description: "The metric to be returned, along with statistics, period, and units. Use this parameter only if this object is retrieving a metric and not performing a math expression on returned data.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromQual("metric_stat"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Label"),
+			},
+		}),
+	}
+}
+
+type MetricDataPoint struct {
+	Id         *string
+	Label      *string
+	StatusCode types.StatusCode
+	Period     *int32
+	Timestamp  time.Time
+	Value      float64
+}
+
+//// LIST FUNCTION
+
+func listCloudWatchMetricDataPoints(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	// check if expression & metric_stat both are available, or both are empty
+	if (d.EqualsQuals["expression"] != nil && d.EqualsQuals["metric_stat"] != nil) || (d.EqualsQuals["expression"] == nil && d.EqualsQuals["metric_stat"] == nil) {
+		return nil, errors.New("please provide either expression or metric_stat in where clause to use this table")
+	}
+
+	metricDataQueries := types.MetricDataQuery{}
+	metricDataQueries.Id = aws.String(d.EqualsQuals["id"].GetStringValue())
+
+	// Limiting the results
+	maxLimit := int32(100800)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 1 {
+				maxLimit = 1
```

Comment:
I don't think we need this check since the steampipe plugin SDK already takes care of this condition.

Reply:
updated